microsoft/electionguard-web-api
# 💻 Add .vscode launch and tasks

Add VS Code launch and tasks so we can run the project from vscode

## To Test

Ensure you have docker installed

On Any Operating system:

### Test the Primary local dev workflow
1. Open the command palette in VS Code
2. `Tasks: Run Task` `build` to build the project.
3. Go to the debug tab and run `.NET Core Launch (web)` Expected: the debugger runs.
4. set a break in `ElectionController.cs#get()`
5. issue a get to `https://localhost:5001/election`
6. expected: The breakpoint hits.  Hit continue.  Expected: you get a result

### Test the Docker local dev workflow
1. Open the command palette in VS Code
2. `Tasks: Run Task` `build docker (debug)`  to build the debug container
4. Go to the debug tab and run `.NET Core Attach (docker)` Expected: the debugger runs.
5. set a break in `ElectionController.cs#get()`
6. issue a get to `https://localhost:5001/election`
7. expected: The breakpoint hits.  Hit continue.  Expected: you get a result

### Test the Docker local release workflow
1. Open the command palette in VS Code
2. `Tasks: Run Task` `build docker (release)`  to build the debug container
4. Go to the debug tab and run `.NET Core Attach (docker)` Expected: the debugger fails to attach.
5. set a break in `ElectionController.cs#get()`
6. issue a get to `https://localhost:5001/election`
7. expected: The breakpoint is never hit.  Expected: you get a result

merge after #4
